object/application.go

```diff
@@ -32,7 +32,7 @@ type Application struct {
 	EnablePassword  bool            `json:"enablePassword"`
 	EnableSignUp    bool            `json:"enableSignUp"`
 	Providers       []*ProviderItem `xorm:"varchar(10000)" json:"providers"`
-	SignupItems     []*SignupItem   `xorm:"varchar(1000)" json:"signupItems"`
+	SignupItems     []*SignupItem   `xorm:"text" json:"signupItems"`
 	OrganizationObj *Organization   `xorm:"-" json:"organizationObj"`
```

Comment:
We have no idea how much data it needs to storage, if we use `varchar(1000)` to storage 2000 char it will be split.

And a normal value in my dev database, this field's value is

```json
[{"name":"ID","visible":false,"required":true,"prompted":false,"rule":"Random"},{"name":"Username","visible":true,"required":true,"prompted":false,"rule":"None"},{"name":"Display name","visible":true,"required":true,"prompted":false,"rule":"None"},{"name":"Password","visible":true,"required":true,"prompted":false,"rule":"None"},{"name":"Confirm password","visible":true,"required":true,"prompted":false,"rule":"None"},{"name":"Email","visible":true,"required":true,"prompted":false,"rule":"None"},{"name":"Phone","visible":true,"required":true,"prompted":false,"rule":"None"},{"name":"Agreement","visible":true,"required":true,"prompted":false,"rule":"None"}]
```

Its length is 661 chars, we cannot make sure it won't be 1001 chars.

> (varchar use) L + 1 bytes if column values require 0 − 255 bytes, L + 2 bytes if values may require more than 255 bytes

> (text use) L + 2 bytes, where L < 2^16

https://dev.mysql.com/doc/refman/5.7/en/storage-requirements.html#data-types-storage-reqs-strings